readthedocs/oauth/services/githubapp.py

```diff
@@ -545,5 +545,11 @@ def post_comment(self, build, comment: str):
         comment = f"{comment_marker}\n{comment}"
         if existing_gh_comment:
             existing_gh_comment.edit(body=comment)
-        else:
+        elif not update_only:
```

Comment:
Seems more explicit to have something like:

```suggestion
        elif comment_on_no_files_changed:
```

The `update_only` concept feels a bit confusing to me, and isn't really what we're trying to solve here?

Reply:
I'll be -1 on renaming this to mention files changed, that's not part of the commenting logic as explained in https://github.com/readthedocs/readthedocs.org/pull/12291#discussion_r2198347290.

Reply:
I don't care about files changes, but we should name it based on the logic, not the inverse. I'd be fine with something like:

```suggestion
        elif create_new:
```

It should describe the actual logic we want, not the inverse of the logic.